app/extensions/auth/security.py

```diff
@@ -0,0 +1,18 @@
+# -*- coding: utf-8 -*-
```

Comment:
I like this new security file.  I'm almost inclined to make a full `security` extension and have it segment functionality away from user authentication, but that seems unnecessarily complicated.  I'm happy to see this code in auth and somewhere we can import from in the future across the project.